launcher/InstanceImportTask.cpp

```diff
@@ -425,8 +433,52 @@ void InstanceImportTask::processFlame()
         m_modIdResolver.reset();
         connect(m_filesNetJob.get(), &NetJob::succeeded, this, [&]()
         {
+            qDebug() << "Finished with blocked mods : " << m_blockedMods.size();
             m_filesNetJob.reset();
-            emitSucceeded();
+            if(!m_blockedMods.empty())
+            {
+                //blocked mods found, we need the slug for displaying.... we need another job :D !
+                auto slugJob = new NetJob("Slug Job", APPLICATION->network());
+                auto slugs = QVector<QByteArray>(m_blockedMods.size());
+                auto index = 0;
+                for(const auto& result: m_blockedMods)
+                {
+                    auto projectId = result.projectId;
+                    slugs[index] = QByteArray();
+                    auto url = QString("https://api.curseforge.com/v1/mods/%1").arg(projectId);
+                    auto dl = Net::Download::makeByteArray(url, &slugs[index]);
+                    slugJob->addNetAction(dl);
+                    index++;
+                }
+                connect(slugJob, &NetJob::succeeded, this, [slugs,this,slugJob]()
+                {
+                    slugJob->deleteLater();
+                    QString text;
+                    auto index = 0;
+                    for(const auto& result: slugs)
+                    {
+                        auto json = QJsonDocument::fromJson(result);
+                        auto slug = json.object().value("data").toObject().value("slug").toString();
+                        auto mod = m_blockedMods[index];
+                        auto link = QString("https://www.curseforge.com/minecraft/mc-mods/%1/download/%2").arg(slug,QString::number(mod.fileId));
```

Comment:
Just a heads up that not all projects in a modpacks manifest are mods and have that url structure, for instance packs can include resource packs which have a base url of https://www.curseforge.com/minecraft/texture-packs. Best thing to do is use the `websiteUrl` in the `links` object for the mod lookup as the basis for the url, then it should be able to handle resourcepacks, and any other type of project that CurseForge has now, or in the future

![image](https://user-images.githubusercontent.com/1723537/170264821-5c4c094a-9b89-49cb-8090-610cc220e9d0.png)

Reply:
Sounds like a cleaner solution to me

Reply:
oh well this triggered on mods only anyways, i don't think polymc even currently downloads non-mods projects currently
I do agree that using that url as a base IS cleaner tho will change